src/script/components/Modals/QualityFeedbackModal/QualityFeedbackModal.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useState} from 'react';
+
+import {amplify} from 'amplify';
+import {container} from 'tsyringe';
+
+import {Button, ButtonVariant, Checkbox, CheckboxLabel} from '@wireapp/react-ui-kit';
+import {WebAppEvents} from '@wireapp/webapp-events';
+
+import {useCallAlertState} from 'Components/calling/useCallAlertState';
+import {ModalComponent} from 'Components/ModalComponent';
+import {RatingListLabel} from 'Components/Modals/QualityFeedbackModal/typings';
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {t} from 'Util/LocalizerUtil';
+import {getLogger} from 'Util/Logger';
+
+import {CALL_QUALITY_FEEDBACK_KEY, CALL_SURVEY_MUTE_INTERVAL, ratingListItems} from './constants';
+import {
+  buttonStyle,
+  buttonWrapper,
+  description,
+  ratingItemBubble,
+  ratingItemHeading,
+  ratingList,
+  title,
+  wrapper,
+} from './QualityFeedbackModal.styles';
+
+import {EventName} from '../../../tracking/EventName';
+import {Segmentation} from '../../../tracking/Segmentation';
+import {UserState} from '../../../user/UserState';
+
+const logger = getLogger('CallQualityFeedback');
+
+interface QualityFeedbackModalProps {
+  userState?: UserState;
+}
+
+export const QualityFeedbackModal = ({userState = container.resolve(UserState)}: QualityFeedbackModalProps) => {
+  const [isChecked, setIsChecked] = useState(false);
+  const {setQualityFeedbackModalShown, qualityFeedBackModalShown} = useCallAlertState();
```

Comment:
Typo:
```suggestion
  const {setQualityFeedbackModalShown, qualityFeedbackModalShown} = useCallAlertState();
```
Maybe we have to update the callee too.